frontend/libs/studio-components/src/components/StudioGridSelector/StudioGridSelector.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { ChangeEvent, useEffect, useRef, useState } from 'react';
+import classes from './StudioGridSelector.module.css';
+import cn from 'classnames';
+import { GridSize } from './types/GridSize';
+
+type OptionData = {
+  value: GridSize;
+  positionX: number;
+};
+
+type StudioGridSelectorProps = {
+  disabled?: boolean;
+  sliderValue?: GridSize;
+  handleSliderChange: (newValue: GridSize) => void;
+};
+
+/**
+ * @component
+ *    A component designed for choosing a value within the range of 1 to 12
+ */
+export const StudioGridSelector = ({
+  disabled = false,
+  sliderValue = 12,
+  handleSliderChange,
+}: StudioGridSelectorProps) => {
+  const [hoverValue, setHoverValue] = useState<number>(0);
+  const [selectedValue, setSelectedValue] = useState<number>(sliderValue);
+  const gridValues: GridSize[] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+  useEffect(() => {
+    setSelectedValue(sliderValue);
+  }, [sliderValue]);
+
+  const optionClassName = (gridValue: number) => {
+    let variableClassName = gridValue > selectedValue ? classes.outside : classes.inside;
+    if (hoverValue > 0) {
+      variableClassName = gridValue > hoverValue ? classes.outside : classes.inside;
+    }
+    return cn(classes.option, variableClassName);
+  };
+
+  const backgroundCss =
+    'linear-gradient(\n' + generateLinearGradient(selectedValue, hoverValue) + ')';
+
+  const inputRef = useRef(null);
+
+  const handleHover = (event) => {
+    const dataListElement = inputRef.current.list;
+    const optionPositionsX: OptionData[] = calculateOptionPositionsX(dataListElement);
+    optionPositionsX.forEach((optionPosX) => {
+      if (optionPosX.positionX < event.clientX) {
+        setHoverValue(optionPosX.value);
```

Comment:
Hvis jeg ikke tar helt feil, vil `setHoverValue` bli kalt flere ganger her helt til den finner riktig verdi. React skal være smart nok til å ikke rerendre komponenten mer enn én gang, men vi bør likevel begrense state-oppdateringer til et minimum for å holde kompleksiteten lav. Hva med å lage en uavhengig funksjon som kun regner ut riktig tall, og kalle den og deretter oppdatere staten?

Reply:
Tenkte det samme, men jeg fant ingen god måte å gjøre det på på stående fot, men ser nå at jeg kan ta reversere listen før jeg tar `.find()` 👍